### PR TITLE
temporal async draft

### PR DESCRIPTION
**Pull request by cgardens**
## What
Proposal on how to get _async_ jobs on temporal.
* Submit job using the scheduler loop as we were before. The submitter is now just responsible for submitting the job and saving the temporal id for the job.
* Add a JobListener that is in charge of tracking the temporal job progress and updating state in scheduler db.
* kills JobRetrier as it is no longer relevant because it's handled by temporal

this PR assumes that synchronous job are happening else where (outside of the scheduler app). that should be possible to do since we don't care about the synchronous jobs writing to the db.

obviously this is not a final state, but it gets us on temporal and off the old thing. once we are in that position we can incrementally move towards decluttering.

### CODE REVIEW

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java [hunk 1]
```diff
@@ -91,146 +84,11 @@ public void run() {
...
-          final long intervalInMinutes = TimeUnit.SECONDS.toMinutes(ScheduleHelpers.getIntervalInSecond(schedule.getSchedule()));
-          frequencyString = intervalInMinutes + " min";
-        }
-        metadata.put("frequency", frequencyString);
-      }
-    }
-    return metadata;
+    // temporal submit.
+    // not a great story on how we recover from this submission failing...
+    final UUID temporalId = temporalClient.submit(job); // maybe need to all submit info from workerrunfactory.
```
Comment (cgardens) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobSubmitter.java:89]: `temporalClient` is whatever that client temporal gives us. obviously will need a bit of massaging.

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracking.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JobTracking {
```
Comment (cgardens) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracking.java:25]: this file just moves all of the segment tracking stuff into its own class out of jobsubmitter. for the sake of figuring out if the approach in the PR is good or not, this class can be ignored.

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobListener.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      LOGGER.info("Completed Job-Submitter...");
+    } catch (Throwable e) {
+      LOGGER.error("Job Submitter Error", e);
+    }
+  }
+
+  @VisibleForTesting
+  void persistProgress(Job job) {
+    final Object temporalJob = temporalClient.getJob(job.getTemporalId());
+    final List<Object> temporalAttempts = temporalClient.getAttempts(job.getTemporalId());
```
Comment (michel-tricot) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobListener.java:87]: does the temporal client supports the concept of attempts or are they just isolated jobs that we assign to attempt?

**Comment by cgardens** [comment]
Ideas in here have been incorporated into our temporal impl.

**Pull request closed without merging**